### ISSUE TITLE
Issue 01: coarse multi-task output contract と trace-level head wrapper を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#362]
# Issue 01: coarse multi-task output contract と trace-level head wrapper を追加する

## 背景

Step 2 では、既存の coarse first-break segmentation に加えて、trace ごとの `pickability_score` を出したい。Step 3 では、同じ coarse model から `coarse_usable_score` も出したい。

現状の coarse pipeline は `model(x) -> Tensor[B, 1, H, W]` を前提としており、`coarse/train.py`, `coarse/infer.py`, `common/loss.py`, QC viewer がすべてこの Tensor を直接使っている。multi-task 化する前に、出力形式の後方互換を保つ薄い契約を入れる。

## ゴール

coarse model が次のどちらでも動くようにする。

```python
# legacy
pred = Tensor[B, 1, H, W]

# multitask
pred = {
    "pick_logits": Tensor[B, 1, H, W],
    "pickability_logits": Tensor[B, H],        # optional
    "coarse_usable_logits": Tensor[B, H],      # optional, Step 3
}
```

既存設定では legacy と同じ挙動を維持する。

## 変更対象の候補

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/build_model.py`
- 新規: `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/output.py`
- 新規: `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/multitask_model.py`
- 必要なら: `packages/seisai-models/src/seisai_models/models/encdec2d.py`

### 1. coarse output helper を追加する

`coarse/output.py` を作り、最低限以下を実装する。

```python
PICK_LOGITS_KEY = "pick_logits"
PICKABILITY_LOGITS_KEY = "pickability_logits"
COARSE_USABLE_LOGITS_KEY = "coarse_usable_logits"

CoarseOutput = torch.Tensor | Mapping[str, torch.Tensor]

def get_pick_logits(output: CoarseOutput) -> torch.Tensor: ...
def get_optional_trace_logits(output: CoarseOutput, key: str) -> torch.Tensor | None: ...
def has_trace_logits(output: CoarseOutput, key: str) -> bool: ...
```

要件:

- legacy Tensor の場合、`get_pick_logits(output)` はその Tensor を返す。
- dict の場合、`pick_logits` を必須にする。
- `pick_logits` は `(B, 1, H, W)` を検証する。
- trace logits は `(B, H)` を検証する。
- エラーメッセージに実際の shape を含める。

### 2. trace-level head wrapper を追加する

`coarse/multitask_model.py` に `CoarseMultiTaskModel` を追加する。

想定構成:

```python
class CoarseMultiTaskModel(nn.Module):
    def __init__(
        self,
        base: EncDec2D,
        *,
        enable_pickability: bool,
        enable_coarse_usable: bool,
        trace_head_hidden_chans: int = 64,
        dropout: float = 0.0,
    ): ...

    def forward(self, x: torch.Tensor) -> dict[str, torch.Tensor]: ...
```

trace-level head は、波形そのものを別モデルに渡すのではなく、coarse encoder/decoder の feature から作る。

推奨実装:

1. `base._encode(x)` と `base.decoder(feats)` を使って decoder feature を得る。
2. `base.seg_head(dec[-1])` から segmentation logits を作る。
3. segmentation logits は既存 `EncDec2D.forward()` と同じく入力 `(H, W)` に bilinear interpolate する。
4. trace head は decoder feature `dec[-1]` を time 方向に pooling し、trace 方向へ 1D convolution / MLP で `B,H` にする。

例:

```python
feat = dec[-1]                       # B,C,h,w
feat_trace = feat.mean(dim=-1)       # B,C,h
feat_trace = F.interpolate(feat_trace, size=H, mode="linear", align_corners=False)
logits = head(feat_trace).squeeze(1) # B,H
```

注意:

- STALTA/AIC/cross-correlation は使わない。
- `EncDec2D.use_tta` は現状 coarse builder で `False` にされているため、まずは TTA 対応を増やさない。
- legacy model の forward は変えない。multi-task 有効時のみ wrapper を使う。

### 3. build_model の後方互換

`build_model()` は legacy flat `encdec2d_kwargs` を受け取ったとき、これまで通り `EncDec2D` を返す。

multi-task 用 signature を受け取ったときだけ `CoarseMultiTaskModel` を返す。

想定 signature:

```python
{
    "encdec2d": {...},
    "trace_heads": {
        "pickability": {"enabled": true},
        "coarse_usable": {"enabled": false},
        "hidden_chans": 64,
        "dropout": 0.0,
    }
}
```

この issue では config loader の実装は Issue 02 に任せてもよいが、`build_model()` 側はこの形式を受け取れるようにしておく。

## 受け入れ条件

- 既存の `build_model(legacy_encdec2d_kwargs)` は `EncDec2D` を返し、既存挙動が変わらない。
- multi-task signature で `build_model()` すると、forward が dict を返す。
- `pick_logits.shape == (B,1,H,W)`。
- pickability 有効時、`pickability_logits.shape == (B,H)`。
- coarse usable 有効時、`coarse_usable_logits.shape == (B,H)`。
- helper `get_pick_logits()` を使うことで、legacy Tensor と dict の両方を処理できる。

## テスト

新規テスト候補:

- `tests/test_fbpick_coarse_multitask_model.py`

テスト内容:

1. legacy build path が Tensor を返す。
2. multi-task build path が dict を返す。
3. `get_pick_logits()` が Tensor/dict 両方で動く。
4. trace logits の shape 不一致で分かりやすい例外が出る。

## 非ゴール

- PU loss の実装はしない。Issue 03 で行う。
- coarse infer NPZ への保存はしない。Issue 04 で行う。
- STALTA/AIC/cross-correlation の特徴量は追加しない。
